[CI] SnapshotShutdownIT testSnapshotShutdownProgressTracker failing

**Build Scans:**
- [elasticsearch-intake #24311 / part1](https://gradle-enterprise.elastic.co/s/en5zkajyj4uts)
- [elasticsearch-pull-request #76628 / part-1](https://gradle-enterprise.elastic.co/s/7gjwqxezpfwja)
- [elasticsearch-periodic #8841 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/ctiw6kdyf5d2a)
- [elasticsearch-pull-request #76525 / part-1](https://gradle-enterprise.elastic.co/s/3tttddt2gu54o)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.snapshots.SnapshotShutdownIT.testSnapshotShutdownProgressTracker" -Dtests.seed=A13DC0F4A8B7C5AD -Dtests.locale=sah-RU -Dtests.timezone=America/Marigot -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.snapshots.SnapshotShutdownIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSnapshotShutdownProgressTracker),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected to see SnapshotShutdownProgressTracker index shard snapshot status messages but did not
Expected: <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 4 failures in test testSnapshotShutdownProgressTracker (2.7% fail rate in 148 executions)
- [main] 2 failures in step part-1 (3.9% fail rate in 51 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (3.9% fail rate in 51 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -524,9 +524,6 @@ tests:
 - class: org.elasticsearch.search.query.VectorIT
   method: testFilteredQueryStrategy
   issue: https://github.com/elastic/elasticsearch/issues/129517
-- class: org.elasticsearch.snapshots.SnapshotShutdownIT
-  method: testSnapshotShutdownProgressTracker
-  issue: https://github.com/elastic/elasticsearch/issues/129752
 - class: org.elasticsearch.xpack.security.SecurityRolesMultiProjectIT
   method: testUpdatingFileBasedRoleAffectsAllProjects
   issue: https://github.com/elastic/elasticsearch/issues/129775

--- a/server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java
+++ b/server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java
@@ -411,6 +411,7 @@ public final class SnapshotShardsService extends AbstractLifecycleComponent impl
                 entry.version(),
                 entry.startTime()
             );
+            snapshotStatus.updateStatusDescription("shard snapshot enqueuing to start");
             startShardSnapshotTaskRunner.enqueueTask(new ActionListener<>() {
                 @Override
                 public void onResponse(Releasable releasable) {
@@ -429,7 +430,6 @@ public final class SnapshotShardsService extends AbstractLifecycleComponent impl
                     assert false : wrapperException; // impossible
                 }
             });
-            snapshotStatus.updateStatusDescription("shard snapshot enqueued to start");
         }
 
         // apply some backpressure by reserving one SNAPSHOT thread for the startup work

